Common/Systems/Synchronization/Handler.cs

```diff
@@ -58,14 +61,19 @@ internal virtual void Receive(BinaryReader reader, byte sender)
 		}
 	}
 
-	public virtual void Load(Mod mod)
+	public void Load(Mod mod)
 	{
-		HandlerForMessage.Add(MessageType, this);
+		Id = (byte)HandlersById.Count;
+		HandlersById.Add(Id, this);
```

Comment:
Assertion to notify us in case we run out of bits, prompting switch to varint32.
```suggestion
		// Switch to 7-bit-encoded Int32 if this panics:
		System.Diagnostics.Debug.Assert(HandlersById.Count < byte.MaxValue);

		Id = (byte)HandlersById.Count;
		HandlersById.Add(Id, this);
```